naptime-graphql/src/test/scala/org/coursera/naptime/ari/graphql/schema/TestingFetcherApi.scala

```diff
@@ -0,0 +1,31 @@
+package org.coursera.naptime.ari.graphql.schema
+
+import com.linkedin.data.DataMap
+import org.coursera.naptime.ResponsePagination
+import org.coursera.naptime.ari.FetcherApi
+import org.coursera.naptime.ari.Request
+import org.coursera.naptime.ari.Response
+import play.api.libs.json.JsString
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+/**
+ * An implementation of FetcherApi that allows one to store data keyed by resource name and ID.
+ */
+case class TestingFetcherApi(rawData: TestingFetcherApi.TestingFetcherApiData) extends FetcherApi {
```

Comment:
to be precise with terminology, let's call this `FakeFetcherApi`.

(the terminology first was introduced in http://xunitpatterns.com/Test%20Double.html and then popularized in https://martinfowler.com/articles/mocksArentStubs.html, specifically,
`Fake objects actually have working implementations, but usually take some shortcut which makes them not suitable for production (an in memory database is a good example).`)

Reply:
Great comment, thanks!